lib/seahorn/BvOpSem2.cc

```diff
@@ -789,6 +800,40 @@ class OpSemMemManager {
     return res;
   }
 
+  Expr storeUnalignedIntToMem(Expr val, PtrTy ptr, Expr mem, unsigned byteSz) {
+    unsigned offsetBits = getByteAlignmentBits();
+    assert(offsetBits != 0);
+
+    Expr res;
+    for (unsigned i = 0; i < byteSz; i++) {
+      PtrTy wordAddress = bv::extract(ptrSzInBits() - 1, offsetBits, ptrAdd(ptr, i));
+      PtrTy byteOffset = bv::extract(offsetBits - 1, 0, ptrAdd(ptr, i));
+      byteOffset = bv::zext(byteOffset, wordSzInBits() - 3);
+      // (x << 3) to get bit offset; zero extend to maintain word size
+      PtrTy bitOffset = bv::concat(byteOffset, bv::bvnum(0, 3, ptr->efac()));
+
+      // existing word at address
+      PtrTy alignedPtr = bv::concat(wordAddress, bv::bvnum(0L, offsetBits, ptr->efac()));
+      Expr existingWord = loadAlignedWordFromMem(alignedPtr, mem);
+
+      // set a byte of existing word to 0, byte should be address aligned
+      Expr lowestByteMask = bv::bvnum(0xff, wordSzInBits(), ptr->efac());
+      Expr addressByteMask = mk<BNOT>(mk<BSHL>(lowestByteMask, bitOffset));
+      existingWord = mk<BAND>(existingWord, addressByteMask);
+
+      unsigned lowBit = i * 8;
+      Expr byteToStore = bv::extract(lowBit + 7, lowBit, val);
+
+      // shift into position for zeroed part of existing word; mask and rewrite
+      Expr shiftedByte = mk<BSHL>(bv::zext(byteToStore, wordSzInBits()), bitOffset);
+      Expr updatedWord = mk<BOR>(existingWord, shiftedByte);
+      res = storeAlignedWordToMem(updatedWord, alignedPtr, mem);
+      mem = res;
+    }
```

Comment:
Add a comment on the basic idea: 
  for each byte i
      1. find word w of i
      2. load w
      3. update a byte in w
      4. write result back to memory
     
So this is quite a lot of unaligned reads and writes. 

This is ok for now. We will improve it as it becomes a performance bottleneck. But a comment about how it is working will be useful for future code reviews.